splice_machine/src/main/java/com/splicemachine/backup/BackupSystemProcedures.java

```diff
@@ -657,4 +665,64 @@ public static void SYSCS_RESTORE_SCHEMA(String destSchema,
         }
     }
 
+    public static void SYSCS_ROLLBACK_DATABASE_TO_TRANSACTION(ResultSet[] resultSets) throws StandardException, SQLException {
+        Connection conn = SpliceAdmin.getDefaultConn();
+        LanguageConnectionContext lcc = conn.unwrap(EmbedConnection.class).getLanguageConnection();
+        long snapshotTxId = lcc.getTransactionExecute().getActiveStateTxId();
+        
+        Txn txn;
+        try {
+            txn = SIDriver.driver().lifecycleManager().beginTransaction();
+            txn = txn.elevateToWritable(Bytes.toBytes("rollback"));
+        } catch (IOException e) {
+            return;
+        }
+        try (SpliceTransactionResourceImpl transactionResource = new SpliceTransactionResourceImpl()) {
+            transactionResource.marshallTransaction(txn);
+
+            IgnoreTxnSupplier ignoreTxn = SIDriver.driver() == null ? null : SIDriver.driver().getIgnoreTxnSupplier();
+            if( ignoreTxn != null && ignoreTxn.shouldIgnore(snapshotTxId) ) {
+                throw new Exception("Already rolled back past "+snapshotTxId+". Cannot roll back to it.");
+            }
+
+            BackupManager backupManager = EngineDriver.driver().manager().getBackupManager();
+            
+            long currentTxId = txn.getTxnId();
+
+            // Set Restore Mode to prevent other workloads from running
+            DDLMessage.DDLChange change = ProtoUtil.createRestoreMode(currentTxId);
+            String changeId = DDLUtils.notifyMetadataChange(change);
+            
+            // Rollback
+            LOG.info("Rolling back to "+snapshotTxId+" from "+currentTxId);
+            backupManager.rollbackDatabase(snapshotTxId, currentTxId);
+            
+            // Finish Restore Mode
+            DDLUtils.finishMetadataChange(changeId);
+
+            // Leave restore mode to allow other workload
+            change = ProtoUtil.createLeaveRestoreMode(currentTxId);
+            changeId = DDLUtils.notifyMetadataChange(change);
+            DDLUtils.finishMetadataChange(changeId);
+
+            UpgradeSystemProcedures.restartOlapServer();
+
+            SpliceAdmin.INVALIDATE_GLOBAL_DICTIONARY_CACHE();
+
+            try {
+                txn.commit();
+            } catch (IOException e) {
+                // ignore
+            }
```

Comment:
Is this needed?

Reply:
Removed in commit https://github.com/splicemachine/spliceengine/pull/5137/commits/a9487906fa1be8d3a76f234dacd4527eedb05da3